File: sdk/src/appcheck.ts
import { OwnFirebaseClient } from './client';
import type { AppCheckToken } from './types';

export class AppCheckSDK extends OwnFirebaseClient {
  /**
   * Exchange a platform attestation (e.g. reCAPTCHA token, Play Integrity token,
   * DeviceCheck assertion) for an OwnFirebase App Check token. The token must be
   * included in subsequent API requests as the `X-App-Check-Token` header when
   * App Check enforcement is enabled on the project.
   */
  async exchangeToken(options: {
    provider: 'recaptcha_v3' | 'recaptcha_enterprise' | 'play_integrity' | 'device_check' | 'debug';
    platform: 'web' | 'android' | 'ios';
    rawToken: string;
  }): Promise<AppCheckToken> {
    return this.request('POST', this.projectUrl('app-check/exchange/'), {
      provider: options.provider,
      platform: options.platform,
      raw_token: options.rawToken,
    });
  }
}

File: sdk/python-sdk/ownfirebase/appcheck.py
"""OwnFirebase App Check SDK."""

from typing import Any, Dict

from .client import OwnFirebaseClient


class AppCheckSDK(OwnFirebaseClient):
    """App attestation service."""

    def exchange_token(self, provider: str, platform: str, raw_token: str) -> Dict[str, Any]:
        """Exchange a platform attestation for an OwnFirebase App Check token.

        provider is one of 'recaptcha_v3', 'recaptcha_enterprise', 'play_integrity',
        'device_check', 'debug'. platform is one of 'web', 'android', 'ios'.
        The returned token must be included in subsequent API requests as the
        `X-App-Check-Token` header when App Check enforcement is enabled.
        """
        return self.request(
            'POST',
            self.project_url('app-check/exchange/'),
            json_data={'provider': provider, 'platform': platform, 'raw_token': raw_token},
        )
